docs/setup_vscode_macos.md

```diff
@@ -0,0 +1,554 @@
+```json
+{
+    "configurations": [
+        {
+            ...
+            "program": "${workspaceFolder}/main.exe",
+            "args": ["train_small.csv", "test_small.csv", "--debug"],
+            ...
+        }
+    ]
+}
+```
+{: data-title="launch.json" data-highlight="6" }
+
+## Debug
+In this section, we'll set a breakpoint, which pauses the debugger.  Then, we'll cover some of the options to continue execution.
+
+<img src="images/vscode_icon_step_over.png" style="vertical-align: text-top; height: 1.25em;" /> **Step Over**
+Run one line of code, stepping _over_ any function calls by running the whole function in one step.
+
+<img src="images/vscode_icon_step_in.png" style="vertical-align: text-top; height: 1.25em;" /> **Step Into**
+Run one line of code, stepping _into_ any function calls to execute them line-by-line.
+
+<img src="images/vscode_icon_step_out.png" style="vertical-align: text-top; height: 1.25em;" /> **Step Out**
+Run the program until it returns from the current function (or until the next breakpoint).
+
+<img src="images/vscode_icon_continue.png" style="vertical-align: text-top; height: 1.25em;" /> **Continue**
+Run the program until the next breakpoint.
+
+### Example code
+
+To get started, copy this example `main.cpp` into your editor.
+```c++
+#include <iostream>
+#include <vector>
+using namespace std;
+
+double sum (const vector<double> &data) {
+  double total = 0;
+  for (size_t i=0; i<data.size(); ++i) {
+    total += data[i];
+  }
+  return total;
+}
+
+int main() {
+  vector<double> data;
+  data.push_back(10);
+  data.push_back(20);
+  data.push_back(30);
+  cout << "sum(data) = " << sum(data) << endl;
+}
+```
+{: data-title="main.cpp" }
+
+### Breakpoint
+Select the file you want to debug.  Set a breakpoint by clicking to the left of a line number.  A breakpoint tells the program to pause.
+
+<img src="images/vscode080.png" width="768px" />
+
+### Run
+Select the debugging pane, then run the debugger.  The program pauses at the breakpoint.  The yellow indicator highlights the next line of code to be run.
+
+<img src="images/vscode090.png" width="768px" />
+
+<div class="primer-spec-callout warning" markdown="1">
+**Pitfall:** Don't forget to compile!
+```console
+$ make main.exe                # With a Makefile
+$ g++ -g main.cpp -o main.exe  # Without a Makefile
+```
+</div>
+
+### Step over
+Click "Step Over" a few times until you reach the highlighted line of code
+
+<img src="images/vscode100.png" width="768px" />
+
+### Inspect
+Hover over a variable to inspect its value.  You can also see values in the VARIABLES pane.
+
+<img src="images/vscode110.png" width="768px" />
+
+<div class="primer-spec-callout warning" markdown="1">
+If you have trouble viewing the contents of a container like this screenshot, see [Pretty Printing STL Containers with `gdb`](setup_gdb.html#pretty-printing-stl-containers-with-gdb).
```

Comment:
I don't think this is relevant to macOS, since it uses lldb rather than gdb.

Reply:
Fixed 9d20b6ba28920b124871251ec8e955bf40f0b43b